### PR TITLE
[REVIEW] Fix issue related to `iloc` and slicing a `DataFrame`

### PR DESCRIPTION
**Pull request by galipremsagar**
Fixes: #5757 

This PR fixes three issues:
1. In `DataFrame.iloc` we are extracting the start, stop locations from slice but not actually slicing the `index` and hence we end up only creating an incorrect RangeIndex, fixed this issue by slicing the existing `RangeIndex` with `start` & `stop`.
2. When we try to slice and empty dataframe which has an index we seem to be ignoring the index altogether - Fixed this as well.
3. In `DataFrame.__repr__` we are incorrectly slicing an empty dataframe aswell into 4 quadrants to concat and convert to pandas for printing - This actually results in non-uniform values of the index when DataFrame is empty. Hence handling this as a special case and added tests for the same.

### CODE REVIEW

**Comment by GPUtester** [comment]
Please update the changelog in order to start CI tests.

View the gpuCI docs [here](https://docs.rapids.ai/gpuci#open-pr-or-update-pr).

File: python/cudf/cudf/core/dataframe.py [hunk 1]
```diff
@@ -1154,6 +1154,12 @@ def _get_renderable_dataframe(self):
 
         if len(self) <= nrows and len(self._data.names) <= ncols:
             output = self.copy(deep=False)
+        elif self.empty and len(self.index) > 0:
+            # Incase of Empty DataFrame with index, Pandas prints
+            # first 100 index values followed by ...
+            # To obtain ... at the end of index list, adding 2 more
+            # values, hence fetching only first 102 values.
+            output = self.head(102)
```
Comment (kkraus14) [inline python/cudf/cudf/core/dataframe.py:1162]: Is this not dependent on an option like `pd.options.display.max_rows` for example?

File: python/cudf/cudf/core/frame.py [hunk 2]
```diff
@@ -546,15 +546,17 @@ def _slice(self, arg):
         keep_index = True
         if self.index is not None and isinstance(self.index, RangeIndex):
             keep_index = False
+            if len(self._data) == 0:
+                result = self._empty_like(keep_index)
+                result._index = self.index[start:stop]
+                return result
```
Comment (kkraus14) [inline python/cudf/cudf/core/frame.py:552]:
```suggestion
            if self._num_columns == 0:
                result = self._empty_like(keep_index)
                result._index = self.index[start:stop]
                return result
            keep_index = False
```

File: python/cudf/cudf/core/dataframe.py (hunk 1 above)
Comment (galipremsagar) [reply python/cudf/cudf/core/dataframe.py:1162]: Surprisingly nope, tried playing around with options like max_rows, max_columns. Changing those values(ranging from 0, 2, 4, 100, 1000) made no difference for index representation incase of an empty dataframe

Comment (galipremsagar) [reply python/cudf/cudf/core/dataframe.py:1162]: I think there is one, `pd.options.display.max_seq_items` changing the code accordingly

Comment (galipremsagar) [reply python/cudf/cudf/core/dataframe.py]: Changed the logic to rely on `max_seq_items`

**Comment by galipremsagar** [comment]
rerun tests

**Pull request merged**